src/Data/Vec.agda

```diff
@@ -41,18 +42,14 @@ module _ {P : A → Set p} (P? : Decidable P) where
 
   filter : ∀ {n} → Vec A n → Vec≤ A n
   filter []       = Vec≤.[]
-  filter (a ∷ as) with does (P? a)
-  ... | true  = a Vec≤.∷ filter as
-  ... | false = ≤-cast (ℕ.n≤1+n _) (filter as)
+  filter (a ∷ as) = if does (P? a) then a Vec≤.∷_ else ≤-cast (ℕ.n≤1+n _) $ filter as
```

Comment:
Hmmm. Clearly for you it is more readable to use `if_then_else_` at higher order, rather than `let` the recursive call in a first order expression... but my instincts/taste would make the opposite choice... and save the import from `Function.Base` for the sake of this one use. 

While my sympathies lie with most of this PR, I think that this choice goes against the ambient 'flavour' of `stdlib` programming (and not just my personal taste ;-))

Reply:
Well, if `let` was implemented properly, I would probably use it more. While I happen to be rather fond of this higher-order pattern, I can easily be convinced to forgo this level of 'cuteness'.

Reply:
Ah! I hadn't realised that `let` is problematic in that way... :-(